presto-main/src/test/java/com/facebook/presto/operator/scalar/TestScalarValidation.java

```diff
@@ -203,6 +205,70 @@ public static long bad(@Nullable @SqlType(StandardTypes.DOUBLE) Double value)
         }
     }
 
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "Method .* has @IsNull used without following a parameter with @SqlType")
+    public void testParameterWithConnectorAndIsNull()
+    {
+        extractScalars(ParameterWithConnectorAndIsNull.class);
+    }
+
+    public static final class ParameterWithConnectorAndIsNull
+    {
+        @ScalarFunction
+        @SqlType(StandardTypes.BIGINT)
+        public static long bad(ConnectorSession session, @IsNull boolean isNull)
+        {
+            return 0;
+        }
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "Method .* has @IsNull used without following a parameter with @SqlType")
+    public void testParameterWithIsNullOnly()
+    {
+        extractScalars(ParameterWithIsNullOnly.class);
+    }
+
+    public static final class ParameterWithIsNullOnly
```

Comment:
Let's name this `ParameterWithOnlyIsNull` (putting `IsNull` at the end matches the others)